File: frontend/src/types/risk.ts
export interface RiskFactors {
  severity: number;
  exploitability: number;
  exposure: number;
  asset_criticality: number;
  confidence: number;
}

export interface RiskAssessment {
  id?: string;
  finding_id: string;
  risk_score: number;
  risk_level: 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW' | 'INFO';
  priority: 'P0' | 'P1' | 'P2' | 'P3';
  factors: RiskFactors;
  explanation?: string;
  recommended_action?: string;
}

export interface RepositoryRecalculateResponse {
  repository_id: string;
  findings_processed: number;
  assessments_created: number;
}

File: backend/app/schemas/risk.py
from uuid import UUID
from typing import Optional, Dict
from pydantic import BaseModel, ConfigDict


class RiskFactorsSchema(BaseModel):
    severity: float
    exploitability: float
    exposure: float
    asset_criticality: float
    confidence: float


class RiskAssessmentResponse(BaseModel):
    id: Optional[UUID] = None
    finding_id: UUID
    risk_score: int
    risk_level: str
    priority: str
    factors: RiskFactorsSchema
    explanation: Optional[str] = None
    recommended_action: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class RepositoryRecalculateResponse(BaseModel):
    repository_id: UUID
    findings_processed: int
    assessments_created: int
